### PR TITLE
Sql db seeding 2

### PR DESCRIPTION
**Pull request by tyler-thee-creator**
Completed seed script to seed 10M primary records in PostgreSQL database:

- 10M in primary table "features"
- 70M in additional table "features_list" which joins to "features" - n:1 relationship
- 50M in additional table "content_grid_feature_items" which joins to "features" - n:1 relationship

Some files may have small changes but you can focus on:

1. sql_seeder.js
2. data_generator.js
3. query_generator.js

### CODE REVIEW

File: write_stream_test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  host: 'localhost',
+  database: 'test',
+  port: 5432,
+});
+
+const createUser = () => {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const age = Math.ceil(Math.random() * 100);
+  const likesHiking = Math.ceil(Math.random() * 10) > 5;
```
Comment (masoncpott) [inline write_stream_test.js]: Lines 21 and 22 could also be random numbers generated using faker if you wanted to keep all of you "fake data generation" consistent. Probably not worth changing at this point but I noticed that a lot of people generate random numbers with built in JS even when they are using faker in other parts of their code.

File: write_stream_test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  return `${firstName},${lastName},${age},${likesHiking}\n`;
+}
+
+const startWriting = (writeStream, encoding, done) => {
+  let i = lines;
+
+  const writing = () => {
+    let canWrite = true;
+
+    do {
```
Comment (masoncpott) [inline write_stream_test.js]: what is "do"?

File: write_stream_test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const fileStream = fs.createReadStream(filePath);
+
+    fileStream.on('error', done);
+    queryStream.on('error', done);
+    queryStream.on('finish', done);
+    fileStream.pipe(queryStream);
+  });
+}
+
+copyToDb(`${__dirname}/test.csv`);
```
Comment (masoncpott) [inline write_stream_test.js]: This whole file is great. The research you did shows.

File: database/data_seeding/sql_seeder.js [hunk 4]
```diff
@@ -1,60 +1,113 @@
...
-
-const { generateData } = require('./data_generator.js');
-const { generateInsertQuery } = require('./query_generator.js');
+const faker = require('faker');
+const fs = require('fs');
+const util = require('util');
+const { Pool } = require('pg');
+const copyFrom  = require('pg-copy-streams').from;
 
+const { userName, password } = require('../../database_configs/sql_database.config.js');
```
Comment (masoncpott) [inline database/data_seeding/sql_seeder.js:6]: So I did some research about this. What we are using a config file for can also be done with environment variables. You can declare a new environment variable that represents your username or your password and then just plug in something like "process.env.USERNAME" or whatever the syntax is. The benefit of doing that is that we won't have to create a config file in VIM when we are deployed to EC2, but in my opinion, creating a config file in VIM isn't that difficult or time consuming.

**Review by masoncpott — APPROVED** [review]
Great job! Your sql_seeder file is really good, I learned a lot from reading it. And I like how you wrote a few functions that were modular and could be used in more than one location.

File: write_stream_test.js (hunk 1 above)
Comment (tyler-thee-creator) [reply write_stream_test.js]: Definitely good to know, thanks!

**Pull request merged**